File: BOJ1254.cpp
#include <iostream>
#include <string>
#include <vector>
#include <algorithm>
using namespace std;
bool pellindrome(string a){
  int start=0,end=a.length()-1;
  while(start<=end){
    if(a[start]!=a[end]){
      return false;
    }
    start++;
    end--;
  }
  return true;
}
string chopper(string temp,int index=0){
string result;
  for(int i=temp.length()-1-index;i>=0;i--){
    result.push_back(temp[i]);
  }
  return result;
}
int pellindromeMaker(string temp){
  vector <int> answer;
  for(int i=0;i<=temp.length();i++){
    string check=temp+chopper(temp,i);
    if(pellindrome(check)){
      answer.push_back(check.length());
    }
  }
  sort(answer.begin(),answer.end());
  return answer[0];
}
int main(void){
  string temp;
  cin>>temp;
  cout<<pellindromeMaker(temp);
  return 0;
}

File: BOJ1254.py
def pallindrome(check):
    start=0
    end=len(check)-1
    while start<=end:
        if check[start]!=check[end]:
            return False
        start+=1
        end-=1
    return True
temp=input()
addString=[temp[i] for i in range(len(temp)-1,-1,-1)]
result=''
for i in addString:
    result+=i
answer=[]
for i in range(len(temp)+1):
    add=result[i:]
    check=temp+add
    if pallindrome(check):
        answer.append(len(check))
print(answer[len(answer)-1])
